<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Graded Assessment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. 

Use the links below to read the data files using the Pandas library. 

* Chicago Census Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Public Schools

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Crime Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

**NOTE:** Ensure you use the datasets available on the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


Execute the below code cell to avoid prettytable default error.


In [2]:
!pip install pandas 

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.7/12.7 MB 111.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.1/16.1 MB 112.6 MB/s eta 0:00:00


In [3]:
import pandas as pd

In [15]:
pd.set_option("display.max_columns", None)

In [16]:
socioeconomic = pd.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01")
socioeconomic.head()

COMMUNITY_AREA_NUMBER COMMUNITY_AREA_NAME  PERCENT_OF_HOUSING_CROWDED  \
0                    1.0         Rogers Park                         7.7   
1                    2.0          West Ridge                         7.8   
2                    3.0              Uptown                         3.8   
3                    4.0      Lincoln Square                         3.4   
4                    5.0        North Center                         0.3   

   PERCENT_HOUSEHOLDS_BELOW_POVERTY  PERCENT_AGED_16__UNEMPLOYED  \
0                              23.6                          8.7   
1                              17.2                          8.8   
2                              24.0                          8.9   
3                              10.9                          8.2   
4                               7.5                          5.2   

   PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA  \
0                                          18.2   
1                                          20.8   
2                                          11.8   
3                                          13.4   
4                                           4.5   

   PERCENT_AGED_UNDER_18_OR_OVER_64  PER_CAPITA_INCOME  HARDSHIP_INDEX  
0                              27.5              23939            39.0  
1                              38.5              23040            46.0  
2                              22.2              35787            20.0  
3                              25.5              37524            17.0  
4                              26.2              57123             6.0

In [17]:
schools = pd.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01")
schools.head()

School_ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   

  Elementary, Middle, or High School         Street_Address     City State  \
0                                 ES        615 W Kemper Pl  Chicago    IL   
1                                 ES  7511 S South Shore Dr  Chicago    IL   
2                                 ES     8010 S Kostner Ave  Chicago    IL   
3                                 ES     4619 S Wolcott Ave  Chicago    IL   
4                                 HS        3630 S Wells St  Chicago    IL   

   ZIP_Code    Phone_Number  \
0     60614  (773) 534-5720   
1     60649  (773) 535-6650   
2     60652  (773) 535-2280   
3     60609  (773) 535-4389   
4     60609  (773) 535-1590   

                                                Link  \
0  http://schoolreports.cps.edu/SchoolProgressRep...   
1  http://schoolreports.cps.edu/SchoolProgressRep...   
2  http://schoolreports.cps.edu/SchoolProgressRep...   
3  http://schoolreports.cps.edu/SchoolProgressRep...   
4  http://schoolreports.cps.edu/SchoolProgressRep...   

                      Network_Manager                  Collaborative_Name  \
0        Fullerton Elementary Network  NORTH-NORTHWEST SIDE COLLABORATIVE   
1           Skyway Elementary Network            SOUTH SIDE COLLABORATIVE   
2           Midway Elementary Network        SOUTHWEST SIDE COLLABORATIVE   
3         Pershing Elementary Network        SOUTHWEST SIDE COLLABORATIVE   
4  Southwest Side High School Network        SOUTHWEST SIDE COLLABORATIVE   

  Adequate_Yearly_Progress_Made_ Track_Schedule CPS_Performance_Policy_Status  \
0                             No       Standard              Not on Probation   
1                             No        Track_E              Not on Probation   
2                             No       Standard              Not on Probation   
3                             No        Track_E              Not on Probation   
4                            NDA       Standard              Not on Probation   

  CPS_Performance_Policy_Level HEALTHY_SCHOOL_CERTIFIED  Safety_Icon  \
0                      Level 1                      Yes  Very Strong   
1                      Level 1                       No      Average   
2                      Level 2                       No       Strong   
3                      Level 1                       No      Average   
4              Not Enough Data                      Yes      Average   

   SAFETY_SCORE Family_Involvement_Icon Family_Involvement_Score  \
0          99.0             Very Strong                       99   
1          54.0                  Strong                       66   
2          61.0                     NDA                      NDA   
3          56.0                 Average                       44   
4          49.0                  Strong                       60   

  Environment_Icon  Environment_Score Instruction_Icon  Instruction_Score  \
0           Strong               74.0           Strong               66.0   
1           Strong               74.0      Very Strong               84.0   
2          Average               50.0             Weak               36.0   
3          Average               45.0             Weak               37.0   
4           Strong               60.0          Average               55.0   

  Leaders_Icon Leaders_Score Teachers_Icon Teachers_Score  \
0         Weak            65        Strong             70   
1         Weak            63        Strong             76   
2         Weak           NDA           NDA            NDA   
3         Weak            65       Average             48   
4         Weak            45       Average     

In [18]:
crime = pd.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01")
crime.head()

ID CASE_NUMBER        DATE                     BLOCK IUCR  \
0   3512276    HK587712  2004-08-28        047XX S KEDZIE AVE  890   
1   3406613    HK456306  2004-06-26  009XX N CENTRAL PARK AVE  820   
2   8002131    HT233595  2011-04-04        043XX S WABASH AVE  820   
3   7903289    HT133522  2010-12-30      083XX S KINGSTON AVE  840   
4  10402076    HZ138551  2016-02-02           033XX W 66TH ST  820   

  PRIMARY_TYPE                    DESCRIPTION          LOCATION_DESCRIPTION  \
0        THEFT                  FROM BUILDING            SMALL RETAIL STORE   
1        THEFT                 $500 AND UNDER                         OTHER   
2        THEFT                 $500 AND UNDER  NURSING HOME/RETIREMENT HOME   
3        THEFT  FINANCIAL ID THEFT: OVER $300                     RESIDENCE   
4        THEFT                 $500 AND UNDER                         ALLEY   

   ARREST  DOMESTIC  BEAT  DISTRICT  WARD  COMMUNITY_AREA_NUMBER FBICODE  \
0   False     False   911         9  14.0                   58.0       6   
1   False     False  1112        11  27.0                   23.0       6   
2   False     False   221         2   3.0                   38.0       6   
3   False     False   423         4   7.0                   46.0       6   
4   False     False   831         8  15.0                   66.0       6   

   X_COORDINATE  Y_COORDINATE  YEAR   LATITUDE  LONGITUDE  \
0     1155838.0     1873050.0  2004  41.807440 -87.703956   
1     1152206.0     1906127.0  2004  41.898280 -87.716406   
2     1177436.0     1876313.0  2011  41.815933 -87.624642   
3     1194622.0     1850125.0  2010  41.743665 -87.562463   
4     1155240.0     1860661.0  2016  41.773455 -87.706480   

                        LOCATION  
0    (41.8074405, -87.703955849)  
1  (41.898279962, -87.716405505)  
2  (41.815933131, -87.624642127)  
3  (41.743665322, -87.562462756)  
4  (41.773455295, -87.706480471)

In [31]:
!pip install ipython-sql prettytable

import prettytable

prettytable.DEFAULT = 'DEFAULT'

### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


Load the `pandas` and `sqlite3` libraries and establish a connection to `FinalDB.db`


In [21]:
!pip install ipython-sql
%load_ext sql

In [24]:
import csv, sqlite3

In [25]:
con = sqlite3.connect("FinalDB.db")
cur = con.cursor()

In [26]:
%sql sqlite:///FinalDB.db

In [27]:
socioeconomic.to_sql("CENSUS_DATA", con, if_exists='replace',index=False, method='multi')

78

In [28]:
schools.to_sql("CHICAGO_PUBLIC_SCHOOLS", con , if_exists='replace', index=False,method='multi')

566

In [30]:
crime.to_sql("CHICAGO_CRIME_DATA", con, if_exists='replace',index=False, method='multi')

533

Load the SQL magic module


Use `Pandas` to load the data available in the links above to dataframes. Use these dataframes to load data on to the database `FinalDB.db` as required tables.


Establish a connection between SQL magic module and the database `FinalDB.db`


You can now proceed to the the following questions. Please note that a graded assignment will follow this lab and there will be a question on each of the problems stated below. It can be from the answer you received or the code you write for this problem. Therefore, please keep a note of both your codes as well as the response you generate.


## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [33]:
%sql select count(*) as "count" from CHICAGO_CRIME_DATA

 * sqlite:///FinalDB.db
Done.


[(533,)]

### Problem 2

##### List community area names and numbers with per capita income less than 11000.


In [35]:
%%sql select COMMUNITY_AREA_NUMBER,COMMUNITY_AREA_NAME  from CENSUS_DATA 
    where PER_CAPITA_INCOME<11000 limit 5

 * sqlite:///FinalDB.db
Done.


[(26.0, 'West Garfield Park'),
 (30.0, 'South Lawndale'),
 (37.0, 'Fuller Park'),
 (54.0, 'Riverdale')]

### Problem 3

##### List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis) 


In [37]:
%sql select * from CHICAGO_CRIME_DATA limit 5

 * sqlite:///FinalDB.db
Done.


[(3512276, 'HK587712', '2004-08-28', '047XX S KEDZIE AVE', '890', 'THEFT', 'FROM BUILDING', 'SMALL RETAIL STORE', 0, 0, 911, 9, 14.0, 58.0, '6', 1155838.0, 1873050.0, 2004, 41.8074405, -87.70395585, '(41.8074405, -87.703955849)'),
 (3406613, 'HK456306', '2004-06-26', '009XX N CENTRAL PARK AVE', '820', 'THEFT', '$500 AND UNDER', 'OTHER', 0, 0, 1112, 11, 27.0, 23.0, '6', 1152206.0, 1906127.0, 2004, 41.89827996, -87.71640551, '(41.898279962, -87.716405505)'),
 (8002131, 'HT233595', '2011-04-04', '043XX S WABASH AVE', '820', 'THEFT', '$500 AND UNDER', 'NURSING HOME/RETIREMENT HOME', 0, 0, 221, 2, 3.0, 38.0, '6', 1177436.0, 1876313.0, 2011, 41.81593313, -87.62464213, '(41.815933131, -87.624642127)'),
 (7903289, 'HT133522', '2010-12-30', '083XX S KINGSTON AVE', '840', 'THEFT', 'FINANCIAL ID THEFT: OVER $300', 'RESIDENCE', 0, 0, 423, 4, 7.0, 46.0, '6', 1194622.0, 1850125.0, 2010, 41.74366532, -87.56246276, '(41.743665322, -87.562462756)'),
 (10402076, 'HZ138551', '2016-02-02', '033XX W 66TH ST', '820', 'THEFT', '$500 AND UNDER', 'ALLEY', 0, 0, 831, 8, 15.0, 66.0, '6', 1155240.0, 1860661.0, 2016, 41.7734553, -87.70648047, '(41.773455295, -87.706480471)')]

In [39]:
%sql select distinct DESCRIPTION from CHICAGO_CRIME_DATA 

 * sqlite:///FinalDB.db
Done.


[('FROM BUILDING',),
 ('$500 AND UNDER',),
 ('FINANCIAL ID THEFT: OVER $300',),
 ('OVER $500',),
 ('RETAIL THEFT',),
 ('AGG: FINANCIAL ID THEFT',),
 ('PURSE-SNATCHING',),
 ('POCKET-PICKING',),
 ('ATTEMPT THEFT',),
 ('SIMPLE',),
 ('DOMESTIC BATTERY SIMPLE',),
 ('AGG PO HANDS NO/MIN INJURY',),
 ('PRO EMP HANDS NO/MIN INJURY',),
 ('AGGRAVATED:KNIFE/CUTTING INSTR',),
 ('AGGRAVATED: OTHER DANG WEAPON',),
 ('AGGRAVATED: HANDGUN',),
 ('AGGRAVATED DOMESTIC BATTERY',),
 ('AGGRAVATED DOMESTIC BATTERY: OTHER DANG WEAPON',),
 ('TO VEHICLE',),
 ('TO CITY OF CHICAGO PROPERTY',),
 ('TO PROPERTY',),
 ('CRIMINAL DEFACEMENT',),
 ('POSS: CANNABIS 30GMS OR LESS',),
 ('POSS: HEROIN(WHITE)',),
 ('MANU/DEL:CANNABIS 10GM OR LESS',),
 ('MANU/DELIVER: HEROIN (WHITE)',),
 ('POSS: CRACK',),
 ('FOUND SUSPECT NARCOTICS',),
 ('MANU/DELIVER: METHAMPHETAMINES',),
 ('POSS: COCAINE',),
 ('ATTEMPT POSSESSION NARCOTICS',),
 ('SOLICIT NARCOTICS ON PUBLICWAY',),
 ('MANU/DELIVER:CRACK',),
 ('TELEPHONE THREAT',),
 ('OBSCENE TELEPHONE CALLS',),
 ('VIOLATE ORDER OF PROTECTION',),
 ('HARASSMENT BY TELEPHONE',),
 ('PROBATION VIOLATION',),
 ('HARASSMENT BY ELECTRONIC MEANS',),
 ('PAROLE VIOLATION',),
 ('FALSE/STOLEN/ALTERED TRP',),
 ('OTHER VEHICLE OFFENSE',),
 ('OTHER CRIME AGAINST PERSON',),
 ('SEX OFFENDER: FAIL REG NEW ADD',),
 ('AGGRAVATED PO: OTHER DANG WEAP',),
 ('UNLAWFUL ENTRY',),
 ('FORCIBLE ENTRY',),
 ('AUTOMOBILE',),
 ('THEFT/RECOVERY: AUTOMOBILE',),
 ('ATT: AUTOMOBILE',),
 ('TRUCK, BUS, MOTOR HOME',),
 ('CREDIT CARD FRAUD',),
 ('THEFT OF LABOR/SERVICES',),
 ('FORGERY',),
 ('THEFT OF LOST/MISLAID PROP',),
 ('FRAUD OR CONFIDENCE GAME',),
 ('THEFT BY LESSEE,MOTOR VEH',),
 ('BOGUS CHECK',),
 ('FINANCIAL IDENTITY THEFT OVER $ 300',),
 ('ILLEGAL USE CASH CARD',),
 ('ARMED: HANDGUN',),
 ('STRONGARM - NO WEAPON',),
 ('VEHICULAR HIJACKING',),
 ('ARMED:KNIFE/CUTTING INSTRUMENT',),
 ('AGGRAVATED',),
 ('AGGRAVATED VEHICULAR HIJACKING',),
 ('TO STATE SUP LAND',),
 ('TO RESIDENCE',),
 ('TO LAND',),
 ('CALL OPERATION',),
 ('SOLICIT ON PUBLIC WAY',),
 ('SOLICIT FOR BUSINESS',),
 ('UNLAWFUL POSS OF HANDGUN',),
 ('UNLAWFUL POSS OTHER FIREARM',),
 ('RECKLESS CONDUCT',),
 ('BOMB THREAT',),
 ('AGG SEX ASSLT OF CHILD FAM MBR',),
 ('SEX ASSLT OF CHILD BY FAM MBR',),
 ('CRIM SEX ABUSE BY FAM MEMBER',),
 ('AGG CRIM SEX ABUSE FAM MEMBER',),
 ('ATTEMPT NON-AGGRAVATED',),
 ('NON-AGGRAVATED',),
 ('AGG CRIMINAL SEXUAL ABUSE',),
 ('OTHER',),
 ('RESIST/OBSTRUCT/DISARM OFFICER',),
 ('ESCAPE',),
 ('GAME/DICE',),
 ('SELL/GIVE/DEL LIQUOR TO MINOR',),
 ('ILLEGAL CONSUMPTION BY MINOR',),
 ('BY FIRE',),
 ('FIRST DEGREE MURDER',),
 ('CHILD ABDUCTION/STRANGER',),
 ('INTIMIDATION',),
 ('CYBERSTALKING',),
 ('OBSCENE MATTER',),
 ('LICENSED PREMISE',),
 ('LOST PASSPORT',),
 ('INTOXICATING COMPOUNDS',),
 ('INVOLUNTARY SERVITUDE',),
 ('FOID - REVOCATION',),
 ('AGG RIT MUT: HANDS/FIST/FEET SERIOUS INJURY',),
 ('NOTIFICATION OF CIVIL NO CONTACT ORDER',),
 ('DOMESTIC VIOLENCE',)]

In [42]:
%%sql SELECT CASE_NUMBER , DESCRIPTION	
FROM CHICAGO_CRIME_DATA 
WHERE DESCRIPTION LIKE '%MINOR%'


 * sqlite:///FinalDB.db
Done.


[('HL266884', 'SELL/GIVE/DEL LIQUOR TO MINOR'),
 ('HK238408', 'ILLEGAL CONSUMPTION BY MINOR')]

### Problem 4

##### List all kidnapping crimes involving a child?


In [55]:
%%sql SELECT *
FROM CHICAGO_CRIME_DATA 
WHERE LOWER(DESCRIPTION) LIKE '%child%' and lower(PRIMARY_TYPE) like "%kidnapping%"

 * sqlite:///FinalDB.db
Done.


[(5276766, 'HN144152', '2007-01-26', '050XX W VAN BUREN ST', '1792', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER', 'STREET', 0, 0, 1533, 15, 29.0, 25.0, '20', 1143050.0, 1897546.0, 2007, 41.87490841, -87.75024931, '(41.874908413, -87.750249307)')]

### Problem 5

##### List the kind of crimes that were recorded at schools. (No repetitions)


In [56]:
%%sql SELECT *
FROM CHICAGO_CRIME_DATA 
WHERE LOWER(LOCATION_DESCRIPTION) LIKE '%school%' 

 * sqlite:///FinalDB.db
Done.


[(4006321, 'HL353697', '2005-05-04', '077XX S BURNHAM AVE', '460', 'BATTERY', 'SIMPLE', 'SCHOOL, PUBLIC, GROUNDS', 0, 0, 421, 4, 7.0, 43.0, '08B', 1196066.0, 1854156.0, 2005, 41.75469107, -87.55703869, '(41.754691074, -87.557038686)'),
 (4430638, 'HL725506', '2005-11-09', '048XX N FRANCISCO AVE', '484', 'BATTERY', 'PRO EMP HANDS NO/MIN INJURY', 'SCHOOL, PUBLIC, BUILDING', 1, 0, 2031, 20, 40.0, 4.0, '08B', 1156348.0, 1932071.0, 2005, 41.96938944, -87.70048881, '(41.96938944, -87.700488807)'),
 (6644618, 'HP716225', '2008-12-04', '030XX S DR MARTIN LUTHER KING JR DR', '460', 'BATTERY', 'SIMPLE', 'SCHOOL, PUBLIC, BUILDING', 0, 0, 2112, 1, 2.0, 35.0, '08B', 1179306.0, 1885032.0, 2008, 41.83981621, -87.61751617, '(41.839816207, -87.617516172)'),
 (2341955, 'HH639427', '2002-09-10', '005XX N WALLER AVE', '460', 'BATTERY', 'SIMPLE', 'SCHOOL, PUBLIC, BUILDING', 0, 0, 1512, 15, 29.0, 25.0, '08B', 1138238.0, 1903181.0, 2002, 41.89045993, -87.76778089, '(41.890459933, -87.767780886)'),
 (11110571, 'JA460432', '2017-10-05', '076XX S HOMAN AVE', '460', 'BATTERY', 'SIMPLE', 'SCHOOL, PUBLIC, GROUNDS', 0, 0, 835, 8, 18.0, 70.0, '08B', 1155024.0, 1853614.0, 2017, 41.75412154, -87.70746025, '(41.754121535, -87.707460248)'),
 (7399281, 'HS200939', '2010-03-10', '053XX W CONGRESS PKWY', '1320', 'CRIMINAL DAMAGE', 'TO VEHICLE', 'SCHOOL, PUBLIC, GROUNDS', 0, 0, 1522, 15, 29.0, 25.0, '14', 1140822.0, 1897164.0, 2010, 41.8739014, -87.7584391, '(41.873901397, -87.758439102)'),
 (3530721, 'HK577020', '2004-08-23', '016XX W JONQUIL TER', '2024', 'NARCOTICS', 'POSS: HEROIN(WHITE)', 'SCHOOL, PUBLIC, GROUNDS', 1, 0, 2422, 24, 49.0, 1.0, '18', 1163893.0, 1951001.0, 2004, 42.0211776, -87.67220843, '(42.021177601, -87.67220843)'),
 (7502426, 'HS305355', '2010-05-13', '035XX S WASHTENAW AVE', '1821', 'NARCOTICS', 'MANU/DEL:CANNABIS 10GM OR LESS', 'SCHOOL, PUBLIC, BUILDING', 1, 0, 913, 9, 12.0, 58.0, '18', 1158944.0, 1880896.0, 2010, 41.82890791, -87.69234919, '(41.828907913, -87.692349187)'),
 (8082600, 'HT315369', '2011-05-26', '032XX W ADAMS ST', '545', 'ASSAULT', 'PRO EMP HANDS NO/MIN INJURY', 'SCHOOL, PUBLIC, GROUNDS', 0, 0, 1124, 11, 28.0, 27.0, '08A', 1154752.0, 1898890.0, 2011, 41.87837031, -87.70724814, '(41.878370307, -87.707248137)'),
 (7174283, 'HR585012', '2009-10-13', '043XX W 79TH ST', '1330', 'CRIMINAL TRESPASS', 'TO LAND', 'SCHOOL, PUBLIC, GROUNDS', 1, 0, 834, 8, 13.0, 70.0, '26', 1148764.0, 1851854.0, 2009, 41.74941446, -87.7304466, '(41.749414464, -87.730446597)'),
 (2075734, 'HH292682', '2002-04-05', '016XX W BRYN MAWR AV', '2850', 'PUBLIC PEACE VIOLATION', 'BOMB THREAT', 'SCHOOL, PRIVATE, BUILDING', 0, 0, 2012, 20, None, None, '26', None, None, 2002, None, None, None),
 (1806896, 'G635735', '2001-10-22', '0000X N HERMITAGE AV', '2850', 'PUBLIC PEACE VIOLATION', 'BOMB THREAT', 'SCHOOL, PUBLIC, BUILDING', 1, 0, 1333, 12, None, None, '26', 1164687.0, 1900646.0, 2001, 41.88298425, -87.67071922, '(41.882984246, -87.670719218)')]

In [58]:
%%sql SELECT distinct PRIMARY_TYPE
FROM CHICAGO_CRIME_DATA 
WHERE LOWER(LOCATION_DESCRIPTION) LIKE '%school%' 

 * sqlite:///FinalDB.db
Done.


[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

In [60]:
%%sql SELECT distinct PRIMARY_TYPE,DESCRIPTION
FROM CHICAGO_CRIME_DATA 
WHERE LOWER(LOCATION_DESCRIPTION) LIKE '%school%'  group by PRIMARY_TYPE,DESCRIPTION

 * sqlite:///FinalDB.db
Done.


[('ASSAULT', 'PRO EMP HANDS NO/MIN INJURY'),
 ('BATTERY', 'PRO EMP HANDS NO/MIN INJURY'),
 ('BATTERY', 'SIMPLE'),
 ('CRIMINAL DAMAGE', 'TO VEHICLE'),
 ('CRIMINAL TRESPASS', 'TO LAND'),
 ('NARCOTICS', 'MANU/DEL:CANNABIS 10GM OR LESS'),
 ('NARCOTICS', 'POSS: HEROIN(WHITE)'),
 ('PUBLIC PEACE VIOLATION', 'BOMB THREAT')]

### Problem 6

##### List the type of schools along with the average safety score for each type.


In [61]:
%sql select  * from CHICAGO_PUBLIC_SCHOOLS limit 5

 * sqlite:///FinalDB.db
Done.


[(610038, 'Abraham Lincoln Elementary School', 'ES', '615 W Kemper Pl', 'Chicago', 'IL', 60614, '(773) 534-5720', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610038.pdf', 'Fullerton Elementary Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 1', 'Yes', 'Very Strong', 99.0, 'Very Strong', '99', 'Strong', 74.0, 'Strong', 66.0, 'Weak', '65', 'Strong', '70', 'Strong', '56', 'Average', '47', '96.00%', 2.0, '96.40%', '95.80%', '80.1', '43.3', '89.6', '84.9', '60.7', '62.6', '81.9', '85.2', '52', '62.4', '66.3', '77.9', 69.7, 64.4, 0.2, 0.9, 'Yellow', 'Green', '67.1', '54.5', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 813, 33, 'NDA', 1171699.458, 1915829.428, 41.92449696, -87.64452163, 7, 'LINCOLN PARK', 43, 18, '(41.92449696, -87.64452163)'),
 (610281, 'Adam Clayton Powell Paideia Community Academy Elementary School', 'ES', '7511 S South Shore Dr', 'Chicago', 'IL', 60649, '(773) 535-6650', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610281.pdf', 'Skyway Elementary Network', 'SOUTH SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 54.0, 'Strong', '66', 'Strong', 74.0, 'Very Strong', 84.0, 'Weak', '63', 'Strong', '76', 'Weak', '46', 'Average', '50', '95.60%', 15.7, '95.30%', '100.00%', '62.4', '51.7', '21.9', '15.1', '29', '42.8', '38.5', '27.4', '44.8', '42.7', '14.1', '34.4', 16.8, 16.5, 0.7, 1.4, 'Green', 'Green', '17.2', '27.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 521, 46, 'NDA', 1196129.985, 1856209.466, 41.76032435, -87.55673627, 43, 'SOUTH SHORE', 7, 4, '(41.76032435, -87.55673627)'),
 (610185, 'Adlai E Stevenson Elementary School', 'ES', '8010 S Kostner Ave', 'Chicago', 'IL', 60652, '(773) 535-2280', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610185.pdf', 'Midway Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 2', 'No', 'Strong', 61.0, 'NDA', 'NDA', 'Average', 50.0, 'Weak', 36.0, 'Weak', 'NDA', 'NDA', 'NDA', 'Average', '47', 'Weak', '41', '95.70%', 2.3, '94.70%', '98.30%', '53.7', '26.6', '38.3', '34.7', '43.7', '57.3', '48.8', '39.2', '46.8', '44', '7.5', '21.9', 18.3, 15.5, -0.9, -1.0, 'Red', 'Red', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 1324, 44, 'NDA', 1148427.165, 1851012.215, 41.74711093, -87.73170248, 70, 'ASHBURN', 13, 8, '(41.74711093, -87.73170248)'),
 (609993, 'Agustin Lara Elementary Academy', 'ES', '4619 S Wolcott Ave', 'Chicago', 'IL', 60609, '(773) 535-4389', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_609993.pdf', 'Pershing Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 56.0, 'Average', '44', 'Average', 45.0, 'Weak', 37.0, 'Weak', '65', 'Average', '48', 'Average', '53', 'Strong', '58', '95.50%', 10.4, '95.80%', '100.00%', '76.9', 'NDA', '26', '24.7', '61.8', '49.7', '39.2', '27.2', '69.7', '60.6', '9.1', '18.2', 11.1, 9.6, 0.9, 2.4, 'Green', 'Green', '42.9', '25', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 556, 42, 'NDA', 1164504.29, 1873959.199, 41.8097569, -87.6721446, 61, 'NEW CITY', 20, 9, '(41.8097569, -87.6721446)'),
 (610513, 'Air Force Academy High School', 'HS', '3630 S Wells St', 'Chicago', 'IL', 60609, '(773) 535-1590', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610513.pdf', 'Southwest Side High School Network', 'SOUTHWEST SIDE COLLABORATIVE', 'NDA', 'Standard', 'Not on Probation', 'Not Enough Data', 'Yes', 'Average', 49.0, 'Strong', '60', 'Strong', 60.0, 'Average', 55.0, 'Weak', '45', 'Average', '54', 'Average', '53', 'Average', '49', '93.30%', 15.6, '96.90%', '100.00%', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', None, None, None, None, 'NDA', 'NDA', 'NDA', 'NDA', '14.6', '14.8', 'NDA', '16', '1.4', 'NDA', 

In [63]:
%sql select  "Elementary, Middle, or High School", avg(SAFETY_SCORE) AS AVG_SAFETY_SCORE from CHICAGO_PUBLIC_SCHOOLS GROUP BY "Elementary, Middle, or High School"

 * sqlite:///FinalDB.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [64]:
%sql select * from CENSUS_DATA limit 10

 * sqlite:///FinalDB.db
Done.


[(1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0),
 (4.0, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0),
 (5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0),
 (6.0, 'Lake View', 1.1, 11.4, 4.7, 2.6, 17.0, 60058, 5.0),
 (7.0, 'Lincoln Park', 0.8, 12.3, 5.1, 3.6, 21.5, 71551, 2.0),
 (8.0, 'Near North Side', 1.9, 12.9, 7.0, 2.5, 22.6, 88669, 1.0),
 (9.0, 'Edison Park', 1.1, 3.3, 6.5, 7.4, 35.3, 40959, 8.0),
 (10.0, 'Norwood Park', 2.0, 5.4, 9.0, 11.5, 39.5, 32875, 21.0)]

In [65]:
%sql select * from CENSUS_DATA order by PERCENT_HOUSEHOLDS_BELOW_POVERTY desc limit 5

 * sqlite:///FinalDB.db
Done.


[(54.0, 'Riverdale', 5.8, 56.5, 34.6, 27.5, 51.5, 8201, 98.0),
 (37.0, 'Fuller Park', 3.2, 51.2, 33.9, 26.6, 44.9, 10432, 97.0),
 (68.0, 'Englewood', 3.8, 46.6, 28.0, 28.5, 42.5, 11888, 94.0),
 (29.0, 'North Lawndale', 7.4, 43.1, 21.2, 27.6, 42.7, 12034, 87.0),
 (27.0, 'East Garfield Park', 8.2, 42.4, 19.6, 21.3, 43.2, 12961, 83.0)]

In [66]:
%sql select * from CENSUS_DATA order by PERCENT_HOUSEHOLDS_BELOW_POVERTY asc limit 5

 * sqlite:///FinalDB.db
Done.


[(9.0, 'Edison Park', 1.1, 3.3, 6.5, 7.4, 35.3, 40959, 8.0),
 (74.0, 'Mount Greenwood', 1.0, 3.4, 8.7, 4.3, 36.8, 34381, 16.0),
 (72.0, 'Beverly', 0.9, 5.1, 8.0, 3.7, 40.5, 39523, 12.0),
 (10.0, 'Norwood Park', 2.0, 5.4, 9.0, 11.5, 39.5, 32875, 21.0),
 (5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0)]

In [105]:
%sql select * from CENSUS_DATA where COMMUNITY_AREA_NUMBER=58

 * sqlite:///FinalDB.db
Done.


[(58.0, 'Brighton Park', 14.4, 23.6, 13.9, 45.1, 39.3, 13089, 84.0)]

### Problem 8

##### Which community area is most crime prone? Display the coumminty area number only.


In [68]:
%%sql SELECT distinct *
FROM CHICAGO_CRIME_DATA limit 5

 * sqlite:///FinalDB.db
Done.


[(3512276, 'HK587712', '2004-08-28', '047XX S KEDZIE AVE', '890', 'THEFT', 'FROM BUILDING', 'SMALL RETAIL STORE', 0, 0, 911, 9, 14.0, 58.0, '6', 1155838.0, 1873050.0, 2004, 41.8074405, -87.70395585, '(41.8074405, -87.703955849)'),
 (3406613, 'HK456306', '2004-06-26', '009XX N CENTRAL PARK AVE', '820', 'THEFT', '$500 AND UNDER', 'OTHER', 0, 0, 1112, 11, 27.0, 23.0, '6', 1152206.0, 1906127.0, 2004, 41.89827996, -87.71640551, '(41.898279962, -87.716405505)'),
 (8002131, 'HT233595', '2011-04-04', '043XX S WABASH AVE', '820', 'THEFT', '$500 AND UNDER', 'NURSING HOME/RETIREMENT HOME', 0, 0, 221, 2, 3.0, 38.0, '6', 1177436.0, 1876313.0, 2011, 41.81593313, -87.62464213, '(41.815933131, -87.624642127)'),
 (7903289, 'HT133522', '2010-12-30', '083XX S KINGSTON AVE', '840', 'THEFT', 'FINANCIAL ID THEFT: OVER $300', 'RESIDENCE', 0, 0, 423, 4, 7.0, 46.0, '6', 1194622.0, 1850125.0, 2010, 41.74366532, -87.56246276, '(41.743665322, -87.562462756)'),
 (10402076, 'HZ138551', '2016-02-02', '033XX W 66TH ST', '820', 'THEFT', '$500 AND UNDER', 'ALLEY', 0, 0, 831, 8, 15.0, 66.0, '6', 1155240.0, 1860661.0, 2016, 41.7734553, -87.70648047, '(41.773455295, -87.706480471)')]

In [84]:
%%sql SELECT  COMMUNITY_AREA_NUMBER, count(COMMUNITY_AREA_NUMBER) as can
FROM CHICAGO_CRIME_DATA  group by COMMUNITY_AREA_NUMBER having COMMUNITY_AREA_NUMBER<>'None' order by can  desc limit 10

 * sqlite:///FinalDB.db
Done.


[(25.0, 43),
 (23.0, 22),
 (68.0, 21),
 (29.0, 16),
 (28.0, 16),
 (8.0, 15),
 (71.0, 14),
 (24.0, 13),
 (67.0, 12),
 (66.0, 12)]

In [90]:
%%sql
SELECT COMMUNITY_AREA_NUMBER, 
       COUNT(COMMUNITY_AREA_NUMBER) AS can, 
       (COUNT(COMMUNITY_AREA_NUMBER) * 1.0 / SUM(COUNT(COMMUNITY_AREA_NUMBER)) OVER ()) * 100 AS percentcan
FROM CHICAGO_CRIME_DATA
GROUP BY COMMUNITY_AREA_NUMBER
HAVING COMMUNITY_AREA_NUMBER <> 'None'
ORDER BY can DESC
LIMIT 10;


 * sqlite:///FinalDB.db
Done.


[(25.0, 43, 8.775510204081632),
 (23.0, 22, 4.489795918367347),
 (68.0, 21, 4.285714285714286),
 (28.0, 16, 3.2653061224489797),
 (29.0, 16, 3.2653061224489797),
 (8.0, 15, 3.061224489795918),
 (71.0, 14, 2.857142857142857),
 (24.0, 13, 2.6530612244897958),
 (66.0, 12, 2.4489795918367347),
 (67.0, 12, 2.4489795918367347)]

In [91]:
%%sql
SELECT count(*)
FROM CHICAGO_CRIME_DATA


 * sqlite:///FinalDB.db
Done.


[(533,)]

In [92]:
%%sql
SELECT count(*)
FROM CHICAGO_CRIME_DATA where COMMUNITY_AREA_NUMBER!='None'

 * sqlite:///FinalDB.db
Done.


[(490,)]


Double-click **here** for a hint

<!--
Query for the 'community area number' that has most number of incidents
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [94]:
%sql select max(HARDSHIP_INDEX) from CENSUS_DATA

 * sqlite:///FinalDB.db
Done.


[(98.0,)]

In [93]:
%sql select * from CENSUS_DATA where HARDSHIP_INDEX in (select max(HARDSHIP_INDEX) from CENSUS_DATA)

 * sqlite:///FinalDB.db
Done.


[(54.0, 'Riverdale', 5.8, 56.5, 34.6, 27.5, 51.5, 8201, 98.0)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [138]:
%%sql select cd.COMMUNITY_AREA_NAME,cc.COMMUNITY_AREA_NUMBER,count(cd.COMMUNITY_AREA_NAME) as cnt
    from CHICAGO_CRIME_DATA cc,CENSUS_DATA cd 
    where cd.COMMUNITY_AREA_NUMBER=cc.COMMUNITY_AREA_NUMBER group by cd.COMMUNITY_AREA_NAME,cc.COMMUNITY_AREA_NUMBER order by cnt desc limit 15

 * sqlite:///FinalDB.db
Done.


[('Austin', 25.0, 43),
 ('Humboldt park', 23.0, 22),
 ('Englewood', 68.0, 21),
 ('Near West Side', 28.0, 16),
 ('North Lawndale', 29.0, 16),
 ('Near North Side', 8.0, 15),
 ('Auburn Gresham', 71.0, 14),
 ('West Town', 24.0, 13),
 ('Chicago Lawn', 66.0, 12),
 ('West Englewood', 67.0, 12),
 ('Greater Grand Crossing', 69.0, 11),
 ('Lake View', 6.0, 11),
 ('Roseland', 49.0, 11),
 ('Brighton Park', 58.0, 10),
 ('New City', 61.0, 10)]

In [106]:
%%sql select cd.COMMUNITY_AREA_NUMBER,cc.COMMUNITY_AREA_NUMBER,cd.COMMUNITY_AREA_NAME 
    from CHICAGO_CRIME_DATA cc,CENSUS_DATA cd 
    where cd.COMMUNITY_AREA_NUMBER=cc.COMMUNITY_AREA_NUMBER and cd.COMMUNITY_AREA_NAME='Brighton Park'

 * sqlite:///FinalDB.db
Done.


[(58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park'),
 (58.0, 58.0, 'Brighton Park')]

In [121]:
%%sql SELECT COMMUNITY_AREA_NAME
FROM CENSUS_DATA
WHERE COMMUNITY_AREA_NUMBER in (SELECT COMMUNITY_AREA_NUMBER FROM CHICAGO_CRIME_DATA GROUP BY COMMUNITY_AREA_NUMBER ORDER BY COUNT(*) DESC limit 10);


 * sqlite:///FinalDB.db
Done.


[('Near North Side',),
 ('Humboldt park',),
 ('West Town',),
 ('Austin',),
 ('Near West Side',),
 ('North Lawndale',),
 ('West Englewood',),
 ('Englewood',),
 ('Auburn Gresham',)]

In [116]:
%%sql SELECT cd.COMMUNITY_AREA_NUMBER, cd.COMMUNITY_AREA_NAME, COUNT(cc.COMMUNITY_AREA_NUMBER)
FROM CHICAGO_CRIME_DATA cc
JOIN CENSUS_DATA cd ON cc.COMMUNITY_AREA_NUMBER = cd.COMMUNITY_AREA_NUMBER
GROUP BY cd.COMMUNITY_AREA_NAME
ORDER BY COUNT(cc.COMMUNITY_AREA_NUMBER) DESC;


 * sqlite:///FinalDB.db
Done.


[(25.0, 'Austin', 43),
 (23.0, 'Humboldt park', 22),
 (68.0, 'Englewood', 21),
 (29.0, 'North Lawndale', 16),
 (28.0, 'Near West Side', 16),
 (8.0, 'Near North Side', 15),
 (71.0, 'Auburn Gresham', 14),
 (24.0, 'West Town', 13),
 (67.0, 'West Englewood', 12),
 (66.0, 'Chicago Lawn', 12),
 (49.0, 'Roseland', 11),
 (6.0, 'Lake View', 11),
 (69.0, 'Greater Grand Crossing', 11),
 (26.0, 'West Garfield Park', 10),
 (61.0, 'New City', 10),
 (58.0, 'Brighton Park', 10),
 (43.0, 'South Shore', 9),
 (46.0, 'South Chicago', 9),
 (31.0, 'Lower West Side', 9),
 (22.0, 'Logan Square', 9),
 (42.0, 'Woodlawn', 8),
 (30.0, 'South Lawndale', 8),
 (16.0, 'Irving Park', 8),
 (38.0, 'Grand Boulevard', 8),
 (27.0, 'East Garfield Park', 8),
 (44.0, 'Chatham', 8),
 (19.0, 'Belmont Cragin', 8),
 (70.0, 'Ashburn', 8),
 (2.0, 'West Ridge', 7),
 (32.0, 'Loop', 7),
 (63.0, 'Gage Park', 7),
 (53.0, 'West Pullman', 6),
 (1.0, 'Rogers Park', 6),
 (21.0, 'Avondale', 6),
 (15.0, 'Portage Park', 5),
 (7.0, 'Lincoln Park', 5),
 (35.0, 'Douglas', 5),
 (48.0, 'Calumet Heights', 5),
 (14.0, 'Albany Park', 5),
 (62.0, 'West Elsdon', 4),
 (3.0, 'Uptown', 4),
 (51.0, 'South Deering', 4),
 (76.0, "O'Hare", 4),
 (5.0, 'North Center', 4),
 (39.0, 'Kenwood', 4),
 (72.0, 'Beverly', 4),
 (45.0, 'Avalon Park', 4),
 (65.0, 'West Lawn', 3),
 (73.0, 'Washington Height', 3),
 (50.0, 'Pullman', 3),
 (10.0, 'Norwood Park', 3),
 (4.0, 'Lincoln Square', 3),
 (20.0, 'Hermosa', 3),
 (17.0, 'Dunning', 3),
 (54.0, 'Riverdale', 2),
 (18.0, 'Montclaire', 2),
 (59.0, 'McKinley Park', 2),
 (11.0, 'Jefferson Park', 2),
 (41.0, 'Hyde Park', 2),
 (56.0, 'Garfield Ridge', 2),
 (37.0, 'Fuller Park', 2),
 (77.0, 'Edgewater', 2),
 (52.0, 'East Side', 2),
 (64.0, 'Clearing', 2),
 (33.0, 'Near South Side', 1),
 (75.0, 'Morgan Park', 1),
 (55.0, 'Hegewisch', 1),
 (12.0, 'Forest Glen', 1),
 (47.0, 'Burnside', 1),
 (60.0, 'Bridgeport', 1)]

## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>
<h4>Abhishek Gagneja</h4>
<!--
## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
|2023-10-18  | 2.6     | Abhishek Gagneja  | Modified instruction set |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |
-->
## <h3 align="center"> © IBM Corporation 2023. All rights reserved. <h3/>
